### DISCUSSION TITLE
Merge 2 AnimationChannel to a single new AnimationChannel?

### DISCUSSION THREAD

**Discussion by chengenzhao** [AlmasB/FXGL#1032] [Ideas]
Hi: 

Just wonder is it possible to provide merge function of two or more different AnimationChannel

e.g.
```java
var anim1 = new AnimationChannel(image, duration, ...);
var anim2 = new AnimationChannel(image, duration, ...);
var anim = new AnmationChannel(anim1, anim2, anim3....);
```

and also it would be great if we could have

```java
texture.playAnimationChannel(anim1)
  .playAnimationChannel(anim2)
  .playAnimationChannel(anim3);
```

besides is it possible to provide reverse order of animation channel 
e.g.
```java
new AnimationChannel(image, 10, 250, 250, Duration.seconds(1), 9, 0);//play frame from the 10th frame to the 1st frame, backward
```

**Comment by AlmasB** [comment]
AnimationChannels cannot be combined in this way, but you can combine multiple images into a single animation channel using an overloaded constructor.

As for playing in reverse, this has given me an idea. Our animation system supports it, but AnimationChannel doesn't use it and instead has its own implementation. If we were to change the implementation from custom to our animation system (from `com.almasb.fxgl.animation`), then playing in reverse and even with an interpolator would be available for free. I'll have a think about whether this is a simple change.
